### PR TITLE
Replace e2e-harness test with go test

### PR DESCRIPTION
**Pull request by sslavic**
Towards https://github.com/giantswarm/giantswarm/issues/4606

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -35,7 +26,13 @@ e2eTest: &e2eTest
     - run:
         name: Run test
         command: |
-          E2E_KUBECONFIG=/e2e/kubeconfig ./e2e-harness test --test-dir=${TEST_DIR}
+          docker run --rm --network host \
+            -v /home/circleci/.go_workspace/src/github.com/giantswarm/kubernetes-nginx-ingress-controller/${TEST_DIR}:/e2e \
```
Comment (corest) [inline .circleci/config.yml]: Do you need to specify this whole path `-v /home/circleci/.go_workspace/src/github.com/giantswarm` ? Don't you have `kubernetes-nginx-ingress-controller` in your working directory already?

Comment (sslavic) [reply .circleci/config.yml]: Yeah, it seem absolute path has to be used, relative didn't work.

Comment (corest) [reply .circleci/config.yml]: Then a nitpick maybe, adding `$(pwd)` instead of this long path

Comment (sslavic) [reply .circleci/config.yml]: Fixed

**Review by corest — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -35,7 +26,13 @@ e2eTest: &e2eTest
     - run:
         name: Run test
         command: |
-          E2E_KUBECONFIG=/e2e/kubeconfig ./e2e-harness test --test-dir=${TEST_DIR}
+          docker run --rm --network host \
+            -v $(pwd)/${TEST_DIR}:/e2e \
+            -v $(pwd):/go/src/github.com/giantswarm/kubernetes-nginx-ingress-controller \
+            -e E2E_KUBECONFIG=/e2e/kubeconfig \
+            -e CIRCLE_SHA1=${CIRCLE_SHA1} \
```
Comment (kopiczko) [inline .circleci/config.yml]: It would be cool to read project.yaml and pass those.

File: .circleci/config.yml [hunk 3]
```diff
@@ -35,7 +26,13 @@ e2eTest: &e2eTest
     - run:
         name: Run test
         command: |
-          E2E_KUBECONFIG=/e2e/kubeconfig ./e2e-harness test --test-dir=${TEST_DIR}
+          docker run --rm --network host \
+            -v $(pwd)/${TEST_DIR}:/e2e \
+            -v $(pwd):/go/src/github.com/giantswarm/kubernetes-nginx-ingress-controller \
```
Comment (kopiczko) [inline .circleci/config.yml]:
```suggestion
            -v $(pwd):/go/src/github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME} \
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -35,7 +26,13 @@ e2eTest: &e2eTest
...
+            -e E2E_KUBECONFIG=/e2e/kubeconfig \
+            -e CIRCLE_SHA1=${CIRCLE_SHA1} \
+            -w /go/src/github.com/giantswarm/kubernetes-nginx-ingress-controller \
```
Comment (kopiczko) [inline .circleci/config.yml]:
```suggestion
            -w /go/src/github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME} \
```

**Review by kopiczko — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (sslavic) [reply .circleci/config.yml]: Will apply same as you did for kubelock - remote project.yaml and consistently name env vars (E2E_*).

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**